synapse/handlers/federation_event.py

```diff
@@ -955,6 +968,18 @@ async def _compute_event_context_with_maybe_missing_prevs(
         seen = await self._store.have_events_in_timeline(prevs)
         missing_prevs = prevs - seen
 
+        # If we've already recently attempted to pull this missing event recently, don't
+        # try it again so soon. Since we have to fetch all of the prev_events, we can
+        # bail early here if we find any to ignore.
+        prevs_to_ignore = await self._store.filter_event_ids_with_pull_attempt_backoff(
+            room_id, missing_prevs
+        )
```

Comment:
I'm wondering if we should only do this check when *we've* tried to pull the event. I'm worried that if we get sent a new event from a server with prevs that we've previously attempted to fetch, we'll 403 that event event if the sending server now has them. This would cause us to drop the new event.

Reply:
AFAICT, we only use `_compute_event_context_with_maybe_missing_prevs` with events that we've pulled*. And `_compute_event_context_with_maybe_missing_prevs` already has a note that it should only be called in scenarios where:

https://github.com/matrix-org/synapse/blob/b6baa46db078c3ef9e6c5751bccb8d2e1c5c5402/synapse/handlers/federation_event.py#L938-L939


Here are the call stacks:


```
_process_pulled_event
_compute_event_context_with_maybe_missing_prevs
```

*does this count? We have events stored as `outliers` already and are trying to de-outlier them. Seems fine.
```
_sync_partial_state_room
update_state_for_partial_state_event
_compute_event_context_with_maybe_missing_prevs
```

Reply:
Oh right, yeah that's fair then. Thanks for checking.